eth/client.go

```diff
@@ -418,7 +419,161 @@ func (c *client) Bond(amount *big.Int, to ethcommon.Address) (*types.Transaction
+	if err != nil {
+		return nil, err
+	}
+	newStake := totalBonded.Add(totalBonded, amount)
+
+	newHints := simulateTranscoderPoolUpdate(to, newStake, transcoders, isFull)
+
+	return c.BondingManagerSession.BondWithHint(
+		amount,
+		to,
+		oldHints.PosPrev,
+		oldHints.PosNext,
+		newHints.PosPrev,
+		newHints.PosNext,
+	)
+}
+
+func (c *client) Unbond(amount *big.Int) (*types.Transaction, error) {
+	sender := c.Account().Address
+
+	// Get delegator
+	delegator, err := c.GetDelegator(sender)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get delegator")
+	}
+
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	// Get total bonded
+	totalBonded, err := c.TranscoderTotalStake(delegator.DelegateAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	newStake := totalBonded.Sub(totalBonded, amount)
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	hints := simulateTranscoderPoolUpdate(delegator.DelegateAddress, newStake, transcoders, isFull)
+
+	return c.UnbondWithHint(amount, hints.PosPrev, hints.PosNext)
+}
+
+func (c *client) RebondFromUnbonded(to ethcommon.Address, unbondingLockID *big.Int) (*types.Transaction, error) {
+	sender := c.Account().Address
+
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	totalBonded, err := c.TranscoderTotalStake(to)
+	if err != nil {
+		return nil, err
+	}
+
+	lock, err := c.GetDelegatorUnbondingLock(sender, unbondingLockID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get unbonding lock")
+	}
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	newStake := totalBonded.Add(totalBonded, lock.Amount)
+
+	hints := simulateTranscoderPoolUpdate(to, newStake, transcoders, isFull)
+
+	return c.RebondFromUnbondedWithHint(to, unbondingLockID, hints.PosPrev, hints.PosNext)
+}
+
+func (c *client) Rebond(unbondingLockID *big.Int) (*types.Transaction, error) {
+	sender := c.Account().Address
+
+	// Get delegator
+	delegator, err := c.GetDelegator(sender)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get delegator")
+	}
+
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	totalBonded := delegator.DelegatedAmount
```

Comment:
We need to use `TranscoderTotalStake(delegator.DelegateAddress)` here to get the delegate's delegated amount (we don't want to use the delegator's delegated amount as is done here).